### PR TITLE
Guard against GetEntByIndex read out of bounds

### PR DESCRIPTION
**Pull request by cpdt**
`0x4000` is used as a maximum for other code reading the same entity array, so seems fine to use here too.

### CODE REVIEW

**Review by emma-miler — APPROVED** [review]

**Comment by cpdt** [comment]
I've had this deployed on my servers for the past few hours, no issues and no crashes.

File: NorthstarDedicatedTest/ExploitFixes.cpp [hunk 1]
```diff
@@ -347,6 +347,20 @@ KHOOK(
...
+// GetEntByIndex (called by ScriptGetEntByIndex) doesn't check for the index being out of bounds when it's
+// above the max entity count. This allows it to be used to crash servers.
+typedef void*(__fastcall* GetEntByIndexType)(int idx);
+GetEntByIndexType GetEntByIndex;
+
+static void* GetEntByIndexHook(int idx)
+{
+	if (idx >= 0x4000)
+	{
+		return nullptr;
```
Comment (RoyalBlue1) [inline NorthstarDedicatedTest/ExploitFixes.cpp:359]: maybe a error message here like "GetEntByIndex {} is out of bounds" or something like that

**Review by RoyalBlue1 — APPROVED** [review]

File: NorthstarDedicatedTest/ExploitFixes.cpp (hunk 1 above)
Comment (emma-miler) [inline NorthstarDedicatedTest/ExploitFixes.cpp:359]:
```suggestion
		spdlog::info("GetEntByIndex {} is out of bounds", idx);
		return nullptr;
```

Comment (GeckoEidechse) [reply NorthstarDedicatedTest/ExploitFixes.cpp:360]: Resolved in 50c919ec9037e9a8bbc9fee888f35f66ce3379de

**Pull request merged**